src/query/api/v1/handler/prometheus/native/read_common.go

```diff
@@ -128,28 +118,9 @@ func read(
 		}
 	}()
 
-	if processErr != nil {
-		// Drain anything remaining
-		drainResultChan(results)
-		return nil, processErr
-	}
-
```

Comment:
Is results channel still a thing?  If so I think we do have to drain it unless the publisher stops publishing correctly.

(Have seen leaked of this kind where publisher to the channel is blocked and the reader has aborted, causing mem leak since publisher never completes)

Reply:
Nice yeah misread this as closing the result channel I removed rather than the internal chan